src/main/java/org/opensearch/sdk/ExtensionSettings.java

```diff
@@ -25,30 +25,25 @@ public class ExtensionSettings {
      */
     public static final String EXTENSION_DESCRIPTOR = "src/test/resources/extension.yml";
 
-    public String getExtensionname() {
-        return extensionname;
+    /**
+     * Jackson requires a default constructor.
+     */
+    public ExtensionSettings() {
```

Comment:
You can make this constructor `private` and Jackson can still use it. This will avoid any confusion with users trying to instantiate the object (in fact, they won't be able to, as the presence of the private default one will prevent Java from adding its own default constructor).  May want to add "ObjectMapper" to the explanation in the javadoc as well, for clarity.